### PR TITLE
leave ledger key untouched when deployMutlisig

### PR DESCRIPTION
**Pull request by mattlockyer**
This PR filters out ledger key from keys that are converted to limited access keys, when user deploys multisig contract.

### CODE REVIEW

**Comment by vgrichina** [comment]
@mattlockyer looks good, but please add regression test

**Review by vgrichina — CHANGES_REQUESTED** [review]
Needs a test demonstrating fixed bug

File: test/test-utils.js [hunk 1]
```diff
@@ -27,8 +27,8 @@ function generateUniqueString(prefix) {
     return `${prefix}-${Date.now()}-${Math.round(Math.random() * 1000000)}`;
 }
 
-async function createAccount(near) {
-    const newAccountName = generateUniqueString('test');
+async function createAccount(near, str = 'test') {
```
Comment (vgrichina) [inline test/test-utils.js:30]: not sure what is the reason for the change? Does it matter for test what is account name prefix?

File: test/account.test.js [hunk 2]
```diff
@@ -223,3 +224,32 @@ describe('with deploy contract', () => {
...
+            data: originalKeys.map(({ public_key: publicKey }, i) => ({ publicKey, kind: i > 0 ? 'ledger' : 'phrase' }))
+        });
+        accountMultisig.postSignedJson = () => ({
+            publicKey: KeyPair.fromRandom('ed25519').getPublicKey()
+        });
+        try {
+            await accountMultisig.deployMultisig([...fs.readFileSync('./test/data/main.wasm')]);
+        } catch (e) {
+            if (e.message.indexOf('Contract method is not found') === -1) {
+                throw(e);
```
Comment (vgrichina) [inline test/account.test.js:247]: nit: it's `throw e;`

File: test/account.test.js [hunk 3]
```diff
@@ -223,3 +224,32 @@ describe('with deploy contract', () => {
...
+
+    test('skip access key if recovery method is "ledger"', async () => {
+        const account = await testUtils.createAccount(nearjs, 'multisig');
+        await account.addKey(KeyPair.fromRandom('ed25519').getPublicKey());
+        const originalKeys = await account.getAccessKeys();
+        const { multisig: { AccountMultisig } } = nearApi;
+        const accountMultisig = new AccountMultisig(nearjs.connection, account.accountId, {});
+        // modifiers to return the key material we want to test against
+        accountMultisig.getRecoveryMethods = () => ({
+            data: originalKeys.map(({ public_key: publicKey }, i) => ({ publicKey, kind: i > 0 ? 'ledger' : 'phrase' }))
```
Comment (vgrichina) [inline test/account.test.js:238]: there should be non-ledger and non-phrase key as well to test that it gets converted properly

File: test/account.test.js [hunk 4]
```diff
@@ -223,3 +224,32 @@ describe('with deploy contract', () => {
...
+        });
+        try {
+            await accountMultisig.deployMultisig([...fs.readFileSync('./test/data/main.wasm')]);
+        } catch (e) {
+            if (e.message.indexOf('Contract method is not found') === -1) {
+                throw(e);
+            }
+        }
+        const currentKeys = await accountMultisig.getAccessKeys();
+        expect(currentKeys[0].access_key.permission).toEqual('FullAccess');
```
Comment (vgrichina) [inline test/account.test.js:251]: this should check that public keys are as desired as well

**Comment by mattlockyer** [comment]
Merged in #453

**Pull request closed without merging**